hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolEncoder.java

```diff
@@ -66,27 +67,25 @@ public HandlerStatus onWrite() throws Exception {
         // sends anything and only swaps itself with the next encoder
         try {
             // First, decoder must receive the protocol
-            if (!isDecoderReceivedProtocol && !channel.isClientMode()) {
+            if (!isDecoderReceivedProtocol) {
```

Comment:
Hi,
I found a race between 
- the `isDecoderReceiveProtocol` check and returning `BLOCKED` state for the pipeline
- `signalProtocolVerified` and `signalWrongProtocol` calls (https://github.com/hazelcast/hazelcast/pull/21391/files#diff-ab56f45101f1b1c199987fa877e77c84c5c60e5a7ef6fbbb38bda7b702d702c9R127-R139) and it makes the pipeline stuck in the blocked state. 

See this execution order:
1) if (!isDecoderReceivedProtocol) check is performed
2) `signalVerifiedProtocol` or `signalWrongProtocol` is called and is completed instantly which including pipeline.wakeup() calls but since the pipeline didn't get into BLOCKED state yet, this wakeup call will have no effect.
3) Then,`BLOCKED` state returned for the pipeline

After that, we won't ever call wakeup() again on this pipeline. So, it will infinitely stuck in the blocked state. 

Adding 250 ms sleep to after this `if(!isDecoderReceivedProtocol)` check makes `AdvancedNetworkIntegrationTest#testMembersReportAllAddresses` fail on my local

Reply:
Maybe we can return clean here?

Reply:
I was completely wrong here. Sorry for the inconvenience.